File: backend/src/main/java/com/back/domain/pet/repository/PetStatusRepository.java
package com.back.domain.pet.repository;

import com.back.domain.pet.entity.Pet;
import com.back.domain.pet.entity.PetStatus;
import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;

public interface PetStatusRepository extends JpaRepository<PetStatus, Long> {
    List<PetStatus> findByPet(Pet testPet);
}

File: backend/src/main/java/com/back/domain/pet/repository/PetStatusRepository.kt
package com.back.domain.pet.repository

import com.back.domain.pet.entity.Pet
import com.back.domain.pet.entity.PetStatus
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface PetStatusRepository : JpaRepository<PetStatus, Long> {
    fun findByPet(pet: Pet): List<PetStatus>
}